### ISSUE TITLE
`no-caret-version-dependencies` does not work with multiple dependencies

### ISSUE THREAD

**Issue by agvctm** [tclindner/npm-package-json-lint#97]
I'm using this in an angular 5.2 project, node 8.12. Library version 3.4.0

When specifying the `no-caret-version-dependencies` to 'error' it will only error out when there is only one dependency. With multiple dependencies, it always returns true. This behavior seems to be verified with unit tests. I changed the following in `no-caret-version-dependencies.test.js` 

```
  context('when package.json has node with an invalid value', function() {
    it('LintIssue object should be returned', function() {
      const packageJsonData = {
        dependencies: {
          'npm-package-json-lint': '^1.0.0'
        }
      };
      const response = lint(packageJsonData, 'error');

      response.lintId.should.equal('no-caret-version-dependencies');
      response.severity.should.equal('error');
      response.node.should.equal('dependencies');
      response.lintMessage.should.equal('You are using an invalid version range. Please do not use ^.');
    });
  });
```

to
```
  context('when package.json has node with an invalid value', function() {
    it('LintIssue object should be returned', function() {
      const packageJsonData = {
        dependencies: {
          'npm-package-json-lint': '^1.0.0',
          'foo': '1.2.3'
        }
      };
      const response = lint(packageJsonData, 'error');

      response.lintId.should.equal('no-caret-version-dependencies');
      response.severity.should.equal('error');
      response.node.should.equal('dependencies');
      response.lintMessage.should.equal('You are using an invalid version range. Please do not use ^.');
    });
  });
```

and get a failing test. It seems like there might be an error in dependency-audit.js:areVersRangesValid, but I can't really see why.

**Comment by agvctm** [comment]
I think the problem is in the rule implementation:
```
const lint = function(packageJsonData, severity) {
  const rangeSpecifier = '^';

  if (packageJsonData.hasOwnProperty(nodeName) && areVersRangesValid(packageJsonData, nodeName, rangeSpecifier)) {
    return new LintIssue(lintId, severity, nodeName, message);
  }

  return true;
};
```

If will only return the LintIssue if areVersRangesValid returns true. But that function will only return true if ALL the dependencies contain the caret. I want an error if ANY of the dependencies have the caret. It looks like the tilde and devDependencies of this have the same behavior. 

If I have time. I'll try to get a PR together it.

**Comment by agvctm** [comment]
It turns out I can use `prefer-absolute-version-dependencies` and `prefer-absolute-version-devDependencies` will accomplish what I need to do. So, I'll close this issue. 

However, I would still consider the behavior of `no-caret|tilde-version-(dev)Dependencies` to be a bug?

Thanks for library though. This will help us to not have to track the `package-lock.json` file.

**Comment by tclindner** [comment]
Hey @agvctm! Thank you for using the project. I'm excited that this project is helping you. I appreciate you putting some much detail in this issue. I've published v3.4.1 that should resolve the issue impacting no-caret|tilde-version-(dev)Dependencies. Let me know if you have any other issues.
